pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -136,25 +97,37 @@ func (o *MercuryTriggerService) RegisterTrigger(ctx context.Context, callback ch
 		return fmt.Errorf("triggerId %s already registered", triggerID)
 	}
 
-	feedIDs, err := o.GetFeedIDs(req)
+	cfg := SubscriberConfig{}
+	err = req.Config.UnwrapTo(&cfg)
 	if err != nil {
 		return err
 	}
 
+	feedIDs := []mercury.FeedID{}
+	for _, feedID := range cfg.FeedIds {
+		mfid, err := mercury.NewFeedID(feedID)
+		if err != nil {
+			return err
+		}
+		feedIDs = append(feedIDs, mfid)
+	}
+
 	if len(feedIDs) == 0 {
 		return errors.New("no feedIDs to register")
 	}
 
-	o.chans[triggerID] = callback
-	o.feedIDsForTriggerID[triggerID] = feedIDs
-	for _, feedID := range feedIDs {
-		// check if we need to initialize the map first
-		if _, ok := o.triggerIDsForFeedID[feedID]; !ok {
-			o.triggerIDsForFeedID[feedID] = make(map[string]bool)
-		}
-		o.triggerIDsForFeedID[feedID][triggerID] = true
+	if int64(cfg.MaxFrequencyMs)%o.tickerResolutionMs != 0 {
+		return fmt.Errorf("MaxFrequencyMs must be a multiple of %d", o.tickerResolutionMs)
 	}
 
+	o.subscribers[triggerID] =
+		&subscriber{
+			ch:         callback,
+			workflowID: wid,
+			config:     cfg,
+		}
+
+	o.chans[triggerID] = callback
```

Comment:
@bolekk Do we need to store the channel twice in maps that are keyed to the same thing? Can we just use o.subscribers everywhere?

Reply:
Yes, good point, I'll change.